libs/template/docs/api/class-directive.md

```diff
@@ -0,0 +1,65 @@
+## ClassDirective
+
+The `rxClass` directive serves a convenient way of using observables for style (class) changes.
+
+Under the hood, it leverages a `RenderStrategy` which in turn takes care of optimizing the change detection
+of your component by passing wanted strategy to `rxClassStrategy`. The `ClassDirective` will apply style (class) changes to the template and manage change detection after it got an initial value.
+So if the incoming `Observable` emits its value lazily (e.g. data coming from `Http`), your style/class will be
+applied lazily as well. This can very positively impact the render performance of your application.
```

Comment:
```suggestion
Under the hood, it leverages a `RenderStrategy` which in turn defines the timing when incoming changes get applied to the directives `ElementRef#nativeElement`. The key difference to the native `ngClass` directive and the main reason to use `rxClass` over `ngClass` is its runtime performance. Due to its push-based nature, the `ClassDirective` is able to apply classes defined as input to its `nativeElement` without the need of running `ChangeDetection` or waiting for `NgZone` to kick in.

For this, it needs an `Observable` as input source.

```typescript
class$ = state$.pipe(
  map(state => state.isActive ? 'active-class' : 'disabled-class' )
)
```

```html
<div [rxClass]="class$"></div>
```

Reply:
why just resolved?

Reply:
I updated this part of the docs in one of my commits already. That's why I marked it as resolved.